packages/valory/skills/abstract_round_abci/common.py

```diff
@@ -130,7 +130,7 @@ def async_act(self) -> Generator:
                 self.context.logger.info(f"Retrieved DRAND values: {observation}.")
 
         if observation:
-            payload = self.payload_class(
+            payload = self.payload_class(  # type: ignore
```

Comment:
because it thinks it's a `Type[BaseTxPayload]` it does not recognize the arguments passed for anything else but the `sender`. Associating this directly via the `AbstractRound` and making accessible via `BaseBehaviour.matching_round` should presumably alleviate this.

Reply:
yes. You know this is where protocols come in nicely. Not the kind we use in our framework for defining speech acts and so on, but the types one uses to define the structure of an interface (as an alternative to an abstract base class).